### PR TITLE
Support elliptic curve keys

### LINKED ISSUE #74: Support EC keys in JCEKS key stores?
The `jceks` package can currently only read RSA keys. We should update it to also support EC keys.

### PR DESCRIPTION
**Pull request by csstaub**
r: @alokmenghrajani @mcpherrinm @christodenny 
Support elliptic curve keys. Closes #74.

### CODE REVIEW

File: jceks/jceks.go [hunk 1]
```diff
@@ -95,10 +104,32 @@ func (e *privateKeyEntry) Recover(password []byte) (*rsa.PrivateKey, error) {
 	if _, err := asn1.Unmarshal(decryptedKey, &pKey); err != nil {
 		return nil, err
 	}
-	if !pKey.Algo.Algorithm.Equal(oidPublicKeyRSA) {
-		return nil, fmt.Errorf("unsupported private-key algorithm: %v", pKey.Algo.Algorithm)
+	if pKey.Algo.Algorithm.Equal(oidPublicKeyRSA) {
```
Comment (mcpherrinm) [inline jceks/jceks.go]: Nit: Can we use a switch here instead?

Comment (csstaub) [inline jceks/jceks.go]: We can't, I believe we have to call `.Equal()` to do the comparison properly.

**Comment by mcpherrinm** [comment]
🚢

**Comment by alokmenghrajani** [comment]
nice

**Pull request merged**